### PR TITLE
LCH-4833: Created commands to add/delete platform sites in the inclus…

### PR DESCRIPTION
**Pull request by ankitsingh0188**

### CODE REVIEW

**Review by Ateee89 — CHANGES_REQUESTED** [review]

File: src/Platform/ACSFPlatform.php [hunk 1]
```diff
@@ -312,4 +335,62 @@ protected function isValidUri(string $uri): bool {
...
+   *   Platform grouping sites placeholder.
+   * @param array $sites
+   *   Platform sites.
+   * @param OutputInterface $output
+   *   Output stream.
+   *
+   * @return array|int
+   *   Array containing filtered list of sites.
+   */
+  protected function filterSitesByGroup(string $group_name, array $sites, OutputInterface $output) {
```
[the remark this answers is not in the document]
Comment (ankitsingh0188) [reply src/Platform/ACSFPlatform.php]: Changed. Thanks.

File: src/Platform/ACSFPlatform.php [hunk 2]
```diff
@@ -196,21 +204,39 @@ public function execute(Command $command, InputInterface $input, OutputInterface
...
+    }
+
+    if ($group_name) {
+      $sites = $this->filterSitesByGroup($group_name, $sites, $output);
+      if (empty($sites)) {
+        return 3;
+      }
+      $sites = array_column($sites, 'uri');
+    }
+    elseif ($uri) {
```
Comment (Ateee89) [inline src/Platform/ACSFPlatform.php:232]: Get rid off `elseif`

File: src/Platform/ACSFPlatform.php [hunk 3]
```diff
@@ -44,6 +47,11 @@ class ACSFPlatform extends PlatformBase implements PlatformSitesInterface, Platf
 
   public const SITEFACTORY_TOKEN = 'acquia.acsf.token';
 
+  public const GROUP_CONFIG_LOCATION = [
```
Comment (Ateee89) [inline src/Platform/ACSFPlatform.php:50]: I have just realised it is not platform specific config but true for all the platforms. Let's move it into PlatformBase and do not define it in multiple places.

**Review by Ateee89 — CHANGES_REQUESTED** [review]
I did not test it manually with ACSF!

File: src/Platform/ACSFPlatform.php [hunk 4]
```diff
@@ -195,21 +198,41 @@ public function execute(Command $command, InputInterface $input, OutputInterface
...
+      $output->writeln('<warning>No sites available. Exiting...</warning>');
+      return 1;
+    }
+
+    $group_name = $input->getOption('group');
+    $uri = $input->getOption('uri');
+
+    if ($group_name && $uri) {
+      $helper = $command->getHelper('question');
+      $question = new ConfirmationQuestion('You have provided both the options, group as well as uri. We will ignore the uri option. Do you want to proceed (y/n)?', TRUE);
```
Comment (amangrover90) [inline src/Platform/ACSFPlatform.php:213]: This is problematic when a command runs internally through PlatformCommand Executioner in which case user isn't shown this question and execution kind of halts without informing anything to the user.

Let's ignore group and only use uri in this case without any choice question.

**Review by amangrover90 — CHANGES_REQUESTED** [review]
Left a comment which impacts command execution.

File: src/Platform/ACSFPlatform.php (hunk 4 above)
Comment (ankitsingh0188) [reply src/Platform/ACSFPlatform.php]: Done.

**Pull request closed without merging**